project/src/main/webapp/gmail.js

```diff
@@ -110,14 +105,57 @@ function appendPre(message) {
     pre.appendChild(textContent);
 }
 
-var question = "What day is the livestream?";
-var arrayList = [];
-var totalObjects = {};
+function getQuery() {
+    const query = document.getElementById("text-input").value;
+    console.log("This is the textarea: " + query);
+    return query;
+}
+
+
+/**
+ * Get Message with given ID.
+ *
+ * @param  {String} messageId ID of Message to get.
+ */
+function getMessage(messageId) {
+    const messageRequest = gapi.client.gmail.users.messages.get({
+        'userId': 'me',
+        'id': messageId,
+        'format': "raw"
+    });
+    messageRequest.execute(response => {
+        //Convert from base64 encoding to text.
+        const decodedEmail = atob(response.raw.replace(/-/g, '+').replace(/_/g, '/'));
+        var hasFormatFlowed = false;
+        const emailBodyStartIndex = decodedEmail.indexOf("Content-Type: text/plain; charset=\"UTF-8\"");
+        const emailBodyEndIndex = (decodedEmail.substring(emailBodyStartIndex)).indexOf("Content-Type: text/html; charset=\"UTF-8\"")
+        var emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex + emailBodyEndIndex);
+        if (emailBodyValue.indexOf("format=flowed") != -1) {
+            hasFormatFlowed = true;
+        }
+        emailBodyValue = emailBodyValue.substring(200);
+        const emailDateStartIndex = decodedEmail.indexOf("Date:");
+        const emailDateEndIndex = decodedEmail.substring(emailDateStartIndex).indexOf("Message-ID");
+        var emailDateValue = decodedEmail.substring(emailDateStartIndex, emailDateStartIndex + emailDateEndIndex);
+        const MessageIDStartIndex = decodedEmail.indexOf("Message-ID:") + 12;
+        const MessageIDEndIndex = decodedEmail.substring(MessageIDStartIndex).indexOf("Subject:") + MessageIDStartIndex;
+        var messageID = decodedEmail.substring(MessageIDStartIndex, MessageIDEndIndex);
```

Comment:
Is it possible to break up all these variable declarations into functions. It seems that each variable relies on each other so breaking it down will make it easier to follow.